pkg/filterengine/filters/object_annotation_checker.go

```diff
@@ -0,0 +1,54 @@
+package filters
+
+import (
+	"github.com/infracloudio/botkube/pkg/events"
+	"github.com/infracloudio/botkube/pkg/filterengine"
+	log "github.com/infracloudio/botkube/pkg/logging"
+	"github.com/infracloudio/botkube/pkg/utils"
+)
+
+// ObjectAnnotationChecker add recommendations to the event object if pod created without any labels
+type ObjectAnnotationChecker struct {
+	Description string
+}
+
+// Register filter
+func init() {
+	filterengine.DefaultFilterEngine.Register(ObjectAnnotationChecker{
+		Description: "Checks if annotations botkube.io/* present in object specs and filters them.",
+	})
+}
+
+// Run filters and modifies event struct
+func (f ObjectAnnotationChecker) Run(object interface{}, event *events.Event) {
+
+	// get objects metadata
+	obj := utils.GetObjectMetaData(object)
+
+	// Check annotations in object
+	if len(obj.Annotations) != 0 {
+		// check annotation botkube.io/disable
+		// annotation botkube.io/disable disables the event notifications from objects
+		if obj.Annotations["botkube.io/disable"] == "true" {
```

Comment:
This filter looks good. Please see if you can make annotation specific actions as internal methods of this filter.
It should be easier to add actions for any future annotations we may desire.

Reply:
resolved it